src/main/java/graphql/normalized/ExecutableNormalizedOperationToAstCompiler.java

```diff
@@ -26,71 +29,81 @@
 import static graphql.language.Argument.newArgument;
 import static graphql.language.Field.newField;
 import static graphql.language.InlineFragment.newInlineFragment;
+import static graphql.language.OperationDefinition.Operation.MUTATION;
+import static graphql.language.OperationDefinition.Operation.QUERY;
+import static graphql.language.OperationDefinition.Operation.SUBSCRIPTION;
 import static graphql.language.SelectionSet.newSelectionSet;
 import static graphql.language.TypeName.newTypeName;
 
 @Internal
 public class ExecutableNormalizedOperationToAstCompiler {
-    public static Document compileToDocument(
-            OperationDefinition.Operation operationKind,
-            String operationName,
-            List<ExecutableNormalizedField> topLevelFields
-    ) {
-        List<Selection<?>> selections = selectionsForNormalizedFields(topLevelFields);
+    public static Document compileToDocument(GraphQLSchema schema,
+                                             OperationDefinition.Operation operationKind,
```

Comment:
Ok now that we have the `schema` again, we can avoid explicitly specifying the `operationKind` and derive it from the `ENF` and `GraphQLSchema` but as I said in the PR desc, my IDE is broken, and that approach required some refactoring which I wasn't prepared to do without the IDE. What a massive pain…